src/main/java/module/TicTacToeService.java

```diff
@@ -0,0 +1,157 @@
+                if(ticTacToeBoard.getBoard(j,i) == 'X')
+                    countValueX++;
+                if(ticTacToeBoard.getBoard(j,i) == 'O')
+                    countValueO++;
+            }
+            if(countValueX == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueX = 0;
+            if(countValueO == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueO = 0;
+        }
+        return false;
+    }
+    boolean rowWiseCheck(){
+        int countValueX = 0;
+        int countValueO = 0;
+
+        for(int i = 0; i < ticTacToeBoard.getCol(); i++){
+            for(int j = 0; j < ticTacToeBoard.getRow(); j++){
+                if(ticTacToeBoard.getBoard(i,j) == 'X')
+                    countValueX++;
+                if(ticTacToeBoard.getBoard(i,j) == 'O')
+                    countValueO++;
+            }
+            if(countValueX == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueX = 0;
+            if(countValueO == ticTacToeBoard.getCol())
+                return true;
+            else
+                countValueO = 0;
+        }
+        return false;
+    }
+    boolean leftDiagonalCheck(){
+        int countValueX =0;
+        int countValueO = 0;
+        int leftDiagonal = 0;
+        for(int i = 0; i < ticTacToeBoard.getRow(); i++){
+            if(ticTacToeBoard.getBoard(i, leftDiagonal) == 'X')
+                countValueX++;
+            if(ticTacToeBoard.getBoard(i, leftDiagonal) == 'O')
+                countValueO ++;
+            leftDiagonal++;
+        }
+        if(countValueO == ticTacToeBoard.getCol() || countValueX == ticTacToeBoard.getCol())
+            return true;
+        return false;
+    }
+    boolean rightDiagonalCheck(){
+        int countValueX = 0;
+        int countValueO =0;
+        int rightDiaogal = ticTacToeBoard.getCol()-1;
+        for(int i = 0; i < ticTacToeBoard.getRow(); i++){
+            if(ticTacToeBoard.getBoard(i, rightDiaogal) == 'X')
+                countValueX++;
+            if(ticTacToeBoard.getBoard(i,rightDiaogal) == 'O')
+                countValueO ++;
+            rightDiaogal--;
+        }
+        if(countValueO == ticTacToeBoard.getCol() || countValueX == ticTacToeBoard.getCol())
+            return true;
+        return false;
+    }
+    boolean checkWinner(){
+        if(rowWiseCheck() || colWiseCheck() || leftDiagonalCheck() || rightDiagonalCheck())
+            return true;
+        else
+            return false;
+    }
+    boolean checkValidMove(int row, int col){
+        if(row < 1 || row > ticTacToeBoard.getRow() || col < 1 || col > ticTacToeBoard.getCol()){
+            return false;
+        }
+        if((ticTacToeBoard.getBoard(row-1, col-1)) != '-'){
+            return false;
+        }
+        return true;
+    }
+    void startGame(List<Pieces> pieces) {
```

Comment:
Different levels of abstraction. I can see the lowest level of abstraction as well here. Those should ideally be in smaller diversions that gets called from here.

Reply:
Can you give me an example, in this function what are the lowest level abstraction?